common/src/warp_runner/ui_adapter/mod.rs

```diff
@@ -62,14 +62,20 @@ impl Message {
 
     // Lazily evaluate if the user is mentioned
     pub fn is_mention_self(&mut self, own: &DID) -> bool {
-        if self.is_mention == 0 {
+        if self.is_mention.is_none() {
             let reg = mention_regex_epattern(&own.to_string());
             match reg.find(&self.inner.lines().join("\n")) {
-                Some(c) => self.is_mention = if c.as_str().starts_with('`') { 1 } else { 2 },
-                None => self.is_mention = 1,
+                Some(c) => {
+                    self.is_mention = if c.as_str().starts_with('`') {
+                        Some(false)
+                    } else {
+                        Some(true)
+                    }
```

Comment:
```suggestion
                    self.is_mention = Some(c.as_str().starts_with('`'));
```

Reply:
Actually, this could be changed so `reg.find` is used

Reply:
actually no. the regex used is a bit special:
regex in rust does not allow lookaheads etc. so to check if a mention is inside codeblocks or not (which should be excluded) i do check
1. since its not possible to check if it is OUTSIDE a codeblock i first detect if its inside
2. if its inside the capture starts with a backtick so its ignored

Reply:
the regex used here is a bit special as we need to check and exclude mentions that are inside codeblocks.
but because its not possible (or without lookaheads which the rust regex does not support) to detect if something is NOT true (e.g. not inside a codeblock) i first check if it IS in a codeblock and filter that match out

Reply:
I was referring to changing it so it would be 

```rust
self.is_mention = reg.find(&self.inner.lines().join("\n")).map(|c| !c.as_str().starts_with('`')).unwrap_or_default();
```